language-support/hs/bindings/src/DA/Ledger/Convert.hs

```diff
@@ -159,6 +161,19 @@ optional = \case
     Left _ -> Nothing
     Right a -> Just a
 
+raiseGetActiveContractsResponse :: LL.GetActiveContractsResponse -> Perhaps (AbsOffset,Maybe WorkflowId,[Event])
+raiseGetActiveContractsResponse = \case
```

Comment:
There's no need for a `case` expression here. The pattern can just come after the function name. Record wild cards might also help to make this a bit less verbose.

Reply:
Are we really happy with the use of `{..}` ?
The way it implicitly brings unknown names into scope seems rather scary.

Reply:
I'm not the biggest fan of `{..}` either but we have plenty of it in our code base already. Instead of
```
f C{x, y} = x+y
```
or
```
f C{..} = x+y
```
I usually prefer
```
f x = x c + y c
```
That avoids the `{..}` and repeating the names at the same time.